test/unittests/test_utils.py

```diff
@@ -571,3 +571,13 @@ def test_detect_cloud_gcp_rc_error(mock_metadata, mock_get_stdout, mock_is_progr
     ])
     assert mock_metadata.call_count == 0
     assert utils._ip_for_cloud is None
+
+
+@mock.patch("re.findall")
```

Comment:
I wouldn't mock the `findall` method. It masks the main functionality of the method.
In this case, I would just set the output of `get_stdout` with a real `ip a` output.
Otherwise, the UT is almost useless